kimchi/src/circuits/polynomials/permutation.rs

```diff
@@ -65,11 +65,14 @@ use std::array;
 
 /// Number of constraints produced by the argument.
 pub const CONSTRAINTS: u32 = 3;
-pub const ZK_ROWS: u64 = 3;
+pub const ZK_ROWS: u64 = 2;
+pub const PERM_FINAL_ACC: u64 = 1;
```

Comment:
as I said in another comment I think ZK_ROWS should be moved (and keep the value 3, but by being computed as `EVALS + 1`)

The PERM_FINAL_ACC should definitely live here, but it needs some comment. Perhaps a link to https://o1-labs.github.io/proof-systems/plonk/zkpm.html and "The permutation constraint should not apply on the final row (not including [ZK_ROWS] otherwise it will wrap around. We only need to check the final value of the permutation accumulator on the final row")

Reply:
(btw I would recommend going through the permutation video I linked, I think this is a good opportunity to learn more about it o.o and ofc the associated explanation in the PLONK paper)

Reply:
looks like PERM_FINAL_ACC isn't used anywhere?

Reply:
Yes, it is not used anywhere.
The reason is that it is already included in `ZK_ROWS` constant(`constants.rs`).
I left this `PERM_FINAL_ACC` since it has comments which could be good guide for devs.